stan/math/prim/mat/meta/vector_seq_view.hpp

```diff
@@ -0,0 +1,85 @@
+#ifndef STAN_MATH_PRIM_MAT_META_VECTOR_SEQ_VIEW_HPP
+#define STAN_MATH_PRIM_MAT_META_VECTOR_SEQ_VIEW_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <vector>
+
+namespace stan {
+  using std::vector;
+  using Eigen::Matrix;
+  using Eigen::Dynamic;
+
+  /**
+   * Provides a low-cost wrapper for situations where you either need an Eigen
+   * Vector or RowVector or a std::vector of them and you want to be agnostic
+   * between those two options. This is similar to scalar_seq_view but instead
+   * of being a sequence-like view over a scalar or seq of scalars, it's a
+   * sequence-like view over a vector or seq of vectors. Notably this version
+   * only allows std::vectors as the sequence type, since we would have
+   * difficulty figuring out which contained type was the container otherwise.
+   *
+   * @tparam T the underlying scalar type inside the underlying Vector
```

Comment:
Code's right, but the doc's wrong.  `T` here is the container type (either an Eigen matrix or standard vector of such), not the underlying scalar.  The class would be very hard to use if it were the underlying scalar type.  Check out the implementation later---it gets the template parameter for the probability function, not the scalar type inside of it.  So in use, we'll have `T` instantiated as some `Matrix<S, R, C>` or `vector<Matrix<S, R, C> >`.